src/test/scala/chiselTests/Inout.scala

```diff
@@ -0,0 +1,132 @@
+abstract class InoutTester extends BasicTester {
+  final val BusValue = "hdeadbeef".U
+
+  final val (cycle, done) = Counter(true.B, 2)
+  when (done) { stop(); stop() } // Double stop for Verilator
+
+  final val writer = Module(new InoutWriterBlackBox)
+  writer.io.in := BusValue
+
+  final def check(reader: Module with InoutReader): Unit =
+    assert(reader.io.out === BusValue)
+}
+
+class InoutSpec extends ChiselFlatSpec {
+  behavior of "Analog stitching"
+
+  it should "work with 2 blackboxes bulk connected" in {
+    assertTesterPasses(new InoutTester {
+      val mod = Module(new InoutReaderBlackBox)
+      mod.io.bus <> writer.io.bus
+      check(mod)
+    }, Seq("/InoutBlackBox.v"))
+  }
+  it should "work with 3 blackboxes connected via wire" in {
+    assertTesterPasses(new InoutTester {
+      val mods = Seq.fill(2)(Module(new InoutReaderBlackBox))
+      val busWire = Wire(Analog(32.W))
+      busWire <> writer.io.bus
+      busWire <> mods(0).io.bus
+      mods(1).io.bus <> busWire // check other bulk connect
+      mods.foreach(check(_))
+    }, Seq("/InoutBlackBox.v"))
+  }
+  // TODO This fails in Firrtl, should this be supported?
+  ignore should "allow Analog wires to be connected" in {
+    assertTesterPasses(new InoutTester {
+      val mod = Module(new InoutReaderBlackBox)
+      val busWire = Seq.fill(2)(Wire(Analog(32.W)))
+      busWire(0) <> writer.io.bus
+      busWire(1) <> mod.io.bus
+      busWire(0) <> busWire(1)
+      check(mod)
+    }, Seq("/InoutBlackBox.v"))
+  }
+  // TODO This should probably actually fail during elaboration rather than during simulation!
+  it should "respect last connect semantics" in {
+    assertTesterFails(new InoutTester {
+      val mods = Seq.fill(2)(Module(new InoutReaderBlackBox))
+      writer.io.bus <> mods(0).io.bus // This connection gets overruled!
+      writer.io.bus <> mods(1).io.bus
+      mods.foreach(check(_))
+    }, Seq("/InoutBlackBox.v"))
+  }
+  it should "work with blackboxes at different levels of the module hierarchy" in {
+    assertTesterPasses(new InoutTester {
+      val mods = Seq(Module(new InoutReaderBlackBox), Module(new InoutReaderWrapper))
+      val busWire = Wire(writer.io.bus)
+      busWire <> writer.io.bus
+      busWire <> mods(0).io.bus
+      busWire <> mods(1).io.bus
+      mods.foreach(check(_))
+    }, Seq("/InoutBlackBox.v"))
+  }
+  // TODO This is not supported in Firrtl, should this be supported?
+  it should "NOT SUPPORT two analog ports in the same module" in {
+    assertTesterFails(new InoutTester {
+      val reader = Module(new InoutReaderBlackBox)
+      val connector = Module(new InoutConnector)
+      connector.io.bus1 <> writer.io.bus
+      reader.io.bus <> connector.io.bus2
+      check(reader)
+    }, Seq("/InoutBlackBox.v"))
+  }
+  // TODO What should happen here?
+  // This creates a wire in order to connect the Analog types
+  // Since it is created within the when scope the connection is actually unconditional
```

Comment:
I'd argue for having this error out because Chisel doesn't have enough information to generate switch hardware, and it creates potential for confusion since chisel essentially is dropping the `when` which is almost certainly not the right (intuitive) thing.